packages/app/src/debug/DebugOverLimit.vue

```diff
@@ -27,15 +28,65 @@ import { useI18n } from '@cy/i18n'
 import { getUrlWithParams } from '@packages/frontend-shared/src/utils/getUrlWithParams'
 import { computed } from '@vue/reactivity'
 
+export type DataRetentionLimitExceeded = { __typename: 'DataRetentionLimitExceeded', dataRetentionDays: number | null }
+
+export type UsageLimitExceeded = | { __typename: 'UsageLimitExceeded', monthlyTests: number | null }
+
+export type CloudRunHidingReason = DataRetentionLimitExceeded | UsageLimitExceeded
+
 const { t } = useI18n()
 
 const props = defineProps<{
+  overLimitReasons: (CloudRunHidingReason | null)[]
   overLimitActionType: OverLimitActionTypeEnum
   overLimitActionUrl: string
+  runAgeDays: number
 }>()
 
 const actionUrl = computed(() => {
   return getUrlWithParams({ url: props.overLimitActionUrl, params: { utmMedium: 'Debug Tab', utmSource: getUtmSource() } })
 })
 
+const overLimitReason = computed<CloudRunHidingReason | null>(() => {
+  // Prefer showing the "Usage Exceeded" messaging if multiple conditions exist
+  return props.overLimitReasons.find(isUsageLimit) || props.overLimitReasons.find(isRetentionLimit) || null
+})
+
+const isPlanAdmin = computed(() => props.overLimitActionType === 'UPGRADE')
+
+const iconClasses = computed(() => {
+  return [
+    'icon-dark-gray-500',
+    {
+      'icon-dark-secondary-jade-400': !isRetentionLimit(overLimitReason.value),
+      'icon-light-secondary-jade-200': !isRetentionLimit(overLimitReason.value),
+      'icon-dark-secondary-orange-400': isRetentionLimit(overLimitReason.value),
+      'icon-light-secondary-orange-200': isRetentionLimit(overLimitReason.value),
+    },
+  ]
+})
+const copy = computed(() => {
+  if (isRetentionLimit(overLimitReason.value)) {
+    return {
+      title: t('debugPage.overLimit.retentionExceededTitle'),
+      message: t('debugPage.overLimit.retentionExceededMessage', { limitDays: overLimitReason.value.dataRetentionDays, runAgeDays: props.runAgeDays }),
+      actionLabel: isPlanAdmin.value ? t('debugPage.overLimit.upgradePlan') : t('debugPage.overLimit.contactAdmin'),
+    }
+  }
+
+  return {
+    title: t('debugPage.overLimit.usageExceededTitle'),
+    message: isPlanAdmin.value ? t('debugPage.overLimit.usageExceededAdminMessage', { numberTests: overLimitReason.value?.monthlyTests || 0 }) : t('debugPage.overLimit.usageExceededUserMessage', { numberTests: overLimitReason.value?.monthlyTests || 0 }),
```

Comment:
Nit: Creating a variable for `numberTests` and using object shorthand here makes this easier to read.